### ISSUE TITLE
`--auto-gen-config` fails to generate `Exclude` list for offenses with partially detected `EnforcedStyle`

### ISSUE THREAD

**Issue by sambostock** [rubocop/rubocop#12574]
While trying out `--auto-gen-config` on `ruby/irb`, I noticed that after running it, there were still offenses if I ran `rubocop` normally. It looks like instead of generating an `Exclude` list for some of the `Naming` cops, instead it would generate `EnforcedStyle: ...` with _the default style_ (which is indeed the most common across the project), but ignore the fact several files have offenses for that style.

I am able to reproduce this as a more isolated case with just `Naming/MethodName`.

--------

## Expected behavior

The generated `.rubocop_todo.yml` should include

```yaml
Naming/MethodName:
  Exclude:
    # ...
```

<details>
<summary><code>rubocop --auto-gen-config --debug</code> (not particularly helpful here)</summary>

```console
% bundle exec rubocop --auto-gen-config --debug
configuration from .rubocop.yml
For /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo: configuration from /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/.rubocop.yml
Inheriting configuration from /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/.rubocop_todo.yml
Default configuration from /Users/sambostock/.gem/ruby/3.2.2/gems/rubocop-1.59.0/config/default.yml
Phase 1 of 2: run Layout/LineLength cop (skipped because Layout/LineLength is disabled)
Phase 2 of 2: run all cops
Inspecting 7 files
Scanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/Gemfile
.Scanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/bad/zero.rb
CScanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/good/five.rb
.Scanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/good/four.rb
.Scanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/good/one.rb
.Scanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/good/three.rb
.Scanning /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/good/two.rb
.

7 files inspected, 2 offenses detected, 1 offense autocorrectable
For /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo: configuration from /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/.rubocop.yml
Inheriting configuration from /Users/sambostock/src/local/rubocop-auto-gen-config-missing-exclude-demo/.rubocop_todo.yml
Created .rubocop_todo.yml.
Finished in 0.10009199986234307 seconds
```

</details>

## Actual behavior

The generated `.rubocop_todo.yml` includes

```yaml
Naming/MethodName:
  EnforcedStyle: snake_case
```

## Steps to reproduce the problem

I have produced both an isolated reproduction repo ([sambostock/rubocop-auto-gen-config-missing-exclude-demo](https://github.com/sambostock/rubocop-auto-gen-config-missing-exclude-demo)), and an even more reduced test case which fails on `master` (as of 94d14fe67f429ffe89dcce2e39e0a3e03d9d5fbf, with the most recent tag being `v1.59.0`).

```diff
diff --git a/spec/rubocop/cli/auto_gen_config_spec.rb b/spec/rubocop/cli/auto_gen_config_spec.rb
index a28359660..7777f70c3 100644
--- a/spec/rubocop/cli/auto_gen_config_spec.rb
+++ b/spec/rubocop/cli/auto_gen_config_spec.rb
@@ -1546,6 +1546,45 @@ RSpec.describe 'RuboCop::CLI --auto-gen-config', :isolated_environment do # rubo
       end
     end
 
+    context '[DO NOT MERGE] sambostock/rubocop-auto-gen-config-missing-exclude-demo (simplified)' do
+      it 'includes an Exclude list' do
+        create_file('.rubocop.yml', <<~YAML)
+          inherit_from: .rubocop_todo.yml
+
+          AllCops:
+            DisabledByDefault: true
+            NewCops: enable
+
+          Naming/MethodName:
+            Enabled: true
+
+          # For comparison
+          Style/FrozenStringLiteralComment:
+            Enabled: true
+        YAML
+
+        create_file('bad.rb', ["def Bad", "end"])
+        create_file("good.rb", ["# frozen_string_literal: true", "", "def good", "end"])
+
+        expect(cli.run(['--auto-gen-config'])).to eq(0)
+        expect(File.readlines('.rubocop_todo.yml')[10..].join)
+          .to eq(<<~YAML)
+            # SupportedStyles: snake_case, camelCase
+            Naming/MethodName:
+              Exclude:
+                - 'bad.rb'
+
+            # Offense count: 1
+            # This cop supports unsafe autocorrection (--autocorrect-all).
+            # Configuration parameters: EnforcedStyle.
+            # SupportedStyles: always, always_true, never
+            Style/FrozenStringLiteralComment:
+              Exclude:
+                - 'bad.rb'
+          YAML
+      end
+    end
+
     it 'can be called when there are no files to inspection' do
       expect(cli.run(['--auto-gen-config'])).to eq(0)
     end
```

## RuboCop version

This fails on `master` as of (as of 94d14fe67f429ffe89dcce2e39e0a3e03d9d5fbf, with the most recent tag being `v1.59.0`), under `ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]`.

Nonetheless,

```console
% bundle exec rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin22]
```

**Comment by jonas054** [comment]
I found the bug and submitted a fix. Thanks @sambostock for the detailed report, which made it easy to debug the problem.
